google/cloud/bigquery/client.py

```diff
@@ -464,6 +466,12 @@ def _create_bqstorage_client(self):
             )
             return None
 
+        try:
+            _verify_bq_storage_version()
```

Comment:
Why not do this check in `Connection.__init__` if the `bqstorage_client` is not None after calling this?

You could even set it to None if the check fails.

Reply:
I see this is also called in `table.py`, so you have the check in one place.  Maybe pass the given value here and do the create-if-none check here and return None if the check fails even if a non-None value is passed in.

Reply:
Sorry, not following, you mean `Connection.__init__` from `dbapi`? That is not necessarily used here, the client methods can hit this path without using `dbapi`.

Reply:
> I see this is also called in `table.py`, so you have the check in one place. Maybe pass the given value here and do the create-if-none check here and return `None` if the check fails even if a non-None value is passed in.

I'll have another look at the convoluted call graph that I drew. :) Might actually work, I'll consider it.